File: leetcode-javascript/javascript/82.js

function ListNode(val, next) {
       this.val = (val===undefined ? 0 : val)
       this.next = (next===undefined ? null : next)
  }
  /**
   * 
   * @param {ListNode} head 
   */
var deleteDuplicates = function(head) {
  if (!head) return head
  let dummy = new ListNode(0, head)
  let cur = dummy

  while(cur.next && cur.next.next) {
    if (cur.next.val === cur.next.next.val) {
      const val = cur.next.val
      while(cur.next && cur.next.val === val) {
        cur.next = cur.next.next
      }
    } else {
      cur = cur.next
    }
  }
  return dummy.next
};

let list = new ListNode(1, new ListNode(1, new ListNode(1, new ListNode(2, new ListNode(3)))))

// console.log(deleteDuplicates(list))/

console.log(deleteDuplicates(new ListNode(1, new ListNode(1))))

File: leetcode-javascript/go/82.go
package main

import "fmt"

type ListNode struct {
	Val  int
	Next *ListNode
}

func deleteDuplicates(head *ListNode) *ListNode {
	dumy := &ListNode{
		0,
		head,
	}
	cur := dumy
	for cur.Next != nil && cur.Next.Next != nil {
		if cur.Next.Val == cur.Next.Next.Val {
			x := cur.Next.Val
			for cur.Next != nil && cur.Next.Val == x {
				cur.Next = cur.Next.Next
			}
		} else {
			cur = cur.Next
		}
	}
	return dumy.Next
}

func main() {
	t := ListNode{
		1,
		&ListNode{
			1,
			&ListNode{
				2,
				nil,
			},
		},
	}
	fmt.Printf("%v", deleteDuplicates(&t))
}
